R/apa_print_anova.R

```diff
@@ -137,34 +137,106 @@ apa_print.anova <- function(
   # if(!is.null(ci)) validate(ci, check_class = "numeric", check_length = 1, check_range = c(0, 1))
   # Add method for levene test
 
-  ellipsis <- list(...)
-  variance_table <- arrange_anova(x)
+  ellipsis <- defaults(
+    list(...)
+    , set.if.null = list(
+      in_paren = FALSE
+    )
+  )
+
   object_heading <- attr(x, "heading")
 
+  if (any(object_heading == "Models:")) {
+    # Model comparisons from lmerTest::anova
+    stop("Model-comparison objects of class 'anova' are not supported.")
+  }
+
+  variance_table <- arrange_anova(x)
+
+
+
+
+
+
   # Some model objects do not contain the necessary information to calculate
   # MSEs and effect-size measures. In such cases, print_anova() needs to be called
   # with arguments es = NULL and mse = FALSE.
 
-  if("apa_variance_table" %in% class(variance_table)) { # car::LeveneTest
+  if("apa_variance_table" %in% class(variance_table)) {
+    # car::LeveneTest ----------------------------------------------------------
     if(length(object_heading) == 1 && grepl("Levene", object_heading)) {
       if(!is.null(ellipsis$es)) stop("Effect sizes are not available for car::LeveneTest-objects.")
       return(print_anova(variance_table, es = NULL, mse = FALSE, ...))
     }
-    # # lmerTest::anova.merModLmerTest ----
-    # if(!is.null(attr(variance_table, "ddf"))) {
-    #   return(print_anova(variance_table, mse = FALSE, es = NULL, ...))
-    # }
-    # # afex::mixed ----
-    # if(any(grepl("Mixed Model", object_heading))) {
-    #   attr(variance_table, "correction") <- unname(
-    #     c(KR = "KR", S = "S", PB = "none", LRT = "none")[attr(x, "method")]
-    #   )
-    #   return(print_anova(variance_table, mse = FALSE, es = NULL, ...))
-    # }
-    # # lmerTest::ranova ----
-    # if(object_heading[1] == "ANOVA-like table for random-effects: Single term deletions") {
-    #   stop("Single-term deletions are not supported, yet.\nVisit https://github.com/crsh/papaja/issues to request support.")
-    # }
+
+    # lmerTest::anova.merModLmerTest -------------------------------------------
```

Comment:
There seems to be some avoidable redundancy here (setting attributes, creating `Effect` column and  removing rownames, call to `canonize()`, `beautify()`, and `glue_apa_results()`).  Could we add one nesting `if` layer to avoid this?

Reply:
I think both cases are almost identical, but I didn't want to nest the code further before switching the other cases to the new pipeline. Thought it is more readable and a better starting point for creating a new "default" case later.